File: Productos/Controllers/ProductosController.cs
using ProductosDB;
using ProductosDB.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace Productos.Controllers
{
    public class ProductosController : Controller
    {
        ProductosDB.Repositorio repo = new ProductosDB.Repositorio();
        // GET: Productos
        public ActionResult Index()
        {
            return View();
        }
        public ActionResult GetProductos()
        {
            Repositorio repo = new Repositorio();
            List<ProductosDB.Entities.Productos> productos = new List<ProductosDB.Entities.Productos>();

            productos = repo.GetProductos();
            return Json(productos, JsonRequestBehavior.AllowGet);
        }

        public ActionResult GetTipoProducto() //aqui llamamos al metodo get de repositorios
        {
            Repositorio repo = new Repositorio();
            List<TipoProducto> list = repo.GetTipoProducto();
            return Json(list, JsonRequestBehavior.AllowGet);
        }

        public ActionResult InsertProducto(ProductosDB.Entities.Productos producto)
        {


            Repositorio repo = new Repositorio();
            producto.CreadoPor = "Admin";
            //producto.ModificadoPor = "Admin";
            Respuesta result = repo.InsertProducto(producto);
            return Json(result, JsonRequestBehavior.AllowGet);
        }

        public ActionResult ActualizarProducto(ProductosDB.Entities.Productos producto)
        {
            try
            {
                producto.ModificadoPor = "Admin";
                Respuesta result = repo.ActualizarProducto(producto);
                return Json(result, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                return Json(new Respuesta { Exitoso = false, Mensaje = "Error al actualizar el producto: " + ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }


        public ActionResult EliminarProducto(int id)
        {
            try
            {
                Respuesta result = repo.EliminarProducto(id);
                return Json(result, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                return Json(new Respuesta { Exitoso = false, Mensaje = "Error al eliminar el producto: " + ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }
    }
}

File: Productos/ControllerJs/productosController.js
productosController.controller('ctrProductos', [
    '$scope',
    'productoService',
    function ($scope, productoService) {
        $scope.newProducto = {};
        $scope.productos = [];
        $scope.TipoProducto = [];

        // Cargar productos y tipo de productos al inicio
        cargarProductos();
        cargarTipoProducto();

        // Función para cargar productos
        function cargarProductos() {
            productoService.GetProductos()
                .then(function (data) {
                    $scope.productos = data;
                })
                .catch(function (error) {
                    console.error('Error al cargar productos:', error);
                });
        }

        // Función para cargar tipo de productos
        function cargarTipoProducto() {
            productoService.GetTipoProducto()
                .then(function (data) {
                    $scope.TipoProducto = data;
                })
                .catch(function (error) {
                    console.error('Error al cargar tipos de productos:', error);
                });
        }

        $scope.GuardarProducto = function () {
            productoService.InsertProducto($scope.newProducto)
                .then(function (result) {
                    if (result.Exitoso) {
                        alertify.success(result.Mensaje);
                        cargarProductos(); // Actualizar lista de productos después de la inserción
                    } else {
                        alertify.error(result.Mensaje);
                    }
                })
                .catch(function (error) {
                    console.error('Error al guardar producto:', error);
                    alertify.error('Error al guardar producto');
                });
        }

        $scope.ActualizarProducto = function (index) {
            var updateProducto = $scope.productos[index];
            productoService.ActualizarProducto(updateProducto)
                .then(function (result) {
                    if (result.Exitoso) {
                        alertify.success(result.Mensaje);
                        cargarProductos(); // Actualizar lista de productos después de la actualización
                    } else {
                        alertify.error(result.Mensaje);
                    }
                })
                .catch(function (error) {
                    console.error('Error al actualizar producto:', error);
                    alertify.error('Error al actualizar producto');
                });
        }

        $scope.EliminarProducto = function (id) {
            productoService.EliminarProducto(id)
                .then(function (result) {
                    if (result.Exitoso) {
                        alertify.success(result.Mensaje);
                        cargarProductos(); // Actualizar lista de productos después de la eliminación
                    } else {
                        alertify.error(result.Mensaje);
                    }
                })
                .catch(function (error) {
                    console.error('Error al eliminar producto:', error);
                    alertify.error('Error al eliminar producto');
                });
        }
    }
]);
